LoongArch-ELF-ABI-EN.adoc

```diff
@@ -0,0 +1,206 @@
+= LoongArch ELF ABI specification
+Loongson Technology Corporation Limited
+V1.00
+
+:encoding: UTF-8
+:frame: top
+:grid: rows
+:valign: middle
+
+== Register Convention
+.Integer Register Convention
+[width="80%",options="header",cols="2m,2m,^5,^3"]
+|====================================================================
+|Name     |Alias        |Meaning             |Preserved across calls
+|$r0      |$zero        |Constant zero       |--
+|$r1      |$ra          |Return address      |No
+|$r2      |$tp          |Thread pointer      |--(Unallocatable)
+|$r3      |$sp          |Stack pointer       |Yes
+|$r4-$r11 |$a0-$a7      |Argument registers  |No
+|$r4-$r5  |$v0-$v1      |Return value        |No
+|$r12-$r20|$t0-$t8      |Temp registers      |No
+|$r21     |$x           |Reserved            |--(Unallocatable)
```

Comment:
Because this register is not meant for use in programs conforming to ABI, maybe we should not give it name?

Reply:
‘x’ can be anything. Never mind, just a temp name.

Reply:
Temporary names can and will persist into the future, please don't do this. I've already seen userland code that saves/restores `$x` in their context switch logic.

Not to mention, the names for the still unpublished LASX registers are `$xrNN`-format... Similarity to RISC-V `$xNN` aside, having people with a RISC-V background continuously struggling to differentiate between `$x0`, `$r0`, `$xr0` and `$x` is not even funny...